analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -257,10 +266,66 @@ public void perform() {
+                        + alignmentModel.getVersion();
+                final ProjectVersionRef pvr = SimpleProjectVersionRef.parse(gav);
+                rootModule.getGav().setPVR(pvr);
+                logger.debug("PVR set to {}", rootModule.getGav().getPVR());
+                modules.add(rootModule);
+                append(builder, "------------------- project {}", originalGav);
+                final String newVer = alignmentModel.getVersion();
+
+                if (!originalVer.equals(newVer)) {
+                    logger.debug("Project version : {} ---> {}", originalVer, newVer);
+                    append(builder, "\tProject version : {} ---> {}", originalVer, newVer);
+                }
+
+                projectDependencies.forEach((key, value) -> {
+                    if (configuration.versionModificationEnabled()) {
+                        final ManipulationModel correspondingModule = alignmentModel.findCorrespondingChild(key);
+                        final String group = correspondingModule.getGroup().isEmpty() ? alignmentModel.getGroup()
+                                : correspondingModule.getGroup();
+                        final String artifact = correspondingModule.getName();
+                        final String ga = group + ":" + artifact;
+                        final String originalModuleGav = ga + ":" + (correspondingModule.getOriginalVersion() == null
+                                ? ""
+                                : correspondingModule.getOriginalVersion());
+                        final String newModuleGav = ga + ":" + correspondingModule.getVersion();
+                        logger.debug("Relocation : {} ---> {}", originalModuleGav, newModuleGav);
+                        append(builder, "\tRelocation : {} ---> {}", originalModuleGav, newModuleGav);
+                        final ModulesItem module = new ModulesItem();
+                        module.getGav().setOriginalGAV(originalModuleGav);
+                        module.getGav().setPVR(SimpleProjectVersionRef.parse(newModuleGav));
+                        modules.add(module);
+                    }
+                });
+
+                if (!nonAligned.isEmpty()) {
+                    final String type = "Dependencies";
+
+                    for (ProjectVersionRef na : nonAligned) {
+                        logger.debug("Non-Aligned {} : {}", type, na);
+                        append(builder, "\tNon-Aligned {} : {}", type, na);
+                    }
+                }
+
+                final String report = builder.toString();
+                logger.info(report);
+                writeReport(project.getRootDir(), report);
+                writeJsonModel(project.getRootDir(), jsonReport);
                 GroovyUtils.runCustomGroovyScript(logger, InvocationStage.LAST, project.getRootDir(), configuration,
                         project.getRootProject(),
                         alignmentModel);
```

Comment:
What happens if the groovy scripting makes further changes? This order does not seem right.

Reply:
You are correct. I meant to have these run last, actually.